packages/odyssey-react-mui/src/labs/NavRail/NavRailApp.tsx

```diff
@@ -0,0 +1,119 @@
+/*!
+ * Copyright (c) 2024-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import styled from "@emotion/styled";
+import { useCallback } from "react";
+import {
+  DesignTokens,
+  useOdysseyDesignTokens,
+} from "../../OdysseyDesignTokensContext";
+import { Tooltip } from "../../Tooltip";
+import { MuiPropsContext, MuiPropsContextType } from "../../MuiPropsContext";
+
+const APP_SIDE_LENGTH = "36px";
+const APP_ICON_SIDE_LENGTH = "32px";
+
+const NavRailAppWrapperComponent = styled("div", {
+  shouldForwardProp: (prop) => prop !== "odysseyDesignTokens",
+})(({ odysseyDesignTokens }: { odysseyDesignTokens: DesignTokens }) => ({
+  margin: "auto",
+  marginBottom: odysseyDesignTokens.Spacing3,
+  width: APP_SIDE_LENGTH,
+  height: APP_SIDE_LENGTH,
+}));
+
+const NavRailAppLinkComponent = styled("a", {
+  shouldForwardProp: (prop) =>
+    !["odysseyDesignTokens", "isSelected"].includes(prop),
+})(
+  ({
+    odysseyDesignTokens,
+    isSelected,
+  }: {
+    odysseyDesignTokens: DesignTokens;
+    isSelected: boolean;
+  }) => ({
+    display: "flex",
+    alignItems: "center",
+    justifyContent: "center",
+    margin: 0,
+    width: APP_SIDE_LENGTH,
+    height: APP_SIDE_LENGTH,
+    backgroundColor: isSelected ? odysseyDesignTokens.HueBlue50 : "transparent",
+    borderRadius: odysseyDesignTokens.BorderRadiusTight,
+    transition: `background-color ${odysseyDesignTokens.TransitionDurationMain}`,
+
+    "&:hover, &:focus": {
+      backgroundColor: odysseyDesignTokens.HueNeutral50,
+    },
+  }),
+);
+
+const NavRailAppImgComponent = styled("img")(() => ({
```

Comment:
Copied from the other PR @bryancunningham-okta https://github.com/okta/odyssey/pull/2415#discussion_r1844488328

> IMHO, It would be great if we could render the SVGs inline, rather than using images here

The plan is to load the SVGs from monolith/CDN (https://github.com/atko-eng/okta-ui/pull/8012 https://github.com/atko-eng/okta-core/pull/103856). The pros of this approach is that any app icon changes/additions/removals don't require an Odyssey version bump since it all comes from the backend. The downside is that we can't do inline SVGs like you suggested.